src/Panel.jsx

```diff
@@ -23,22 +35,29 @@ var Panel = React.createClass({
     };
   },
 
-  handleSelect: function (e) {
-    if (this.props.onSelect) {
-      this._isChanging = true;
-      this.props.onSelect(this.props.eventKey);
-      this._isChanging = false;
+  handleSelect: function(e){
+    if(this.props.onSelect) {
+      this.props.onSelect(e, this.props.eventKey)
+    } else {
+      this.handleClickEvent(e);
+      this.handleToggle();
     }
+  },
 
+  handleClickEvent: function(e){
     e.preventDefault();
+  },
 
-    this.setState({
-      expanded: !this.state.expanded
-    });
+  handleToggle: function(){
```

Comment:
I guess it could be possible to move the `componentDidUpdate` code to the mixin.  The `Panel` would still be responsible for setting `expanded=true` and any component that used the mixin would be stuck with using the `expanded` property.  That's probably not a huge deal though.